lib/package.py

```diff
@@ -161,6 +161,9 @@ def _download_source(self, build_dir):
         An alternative to just execute a given command to obtain sources.
         """
         utils.run_command(self.download_source, cwd=build_dir)
+        # automatically append tar.gz if expects_source has no extension
+        if self.expects_source == self.name:
+            self.expects_source = "%s.tar.gz" % self.name
```

Comment:
Appending the extension was never needed here, why do we need it now?
I see the `BuildManager` calls `prepare_packages()`, which downloads the source code.
The `BuildManager` then proceeds to build, which reaches the `Mock` class `build()`, `_prepare()` and `_prepare_archive` methods, and finally `GitRepository.archive()`. This method then uses the `expects_source`, which was passed as the parameter `archive_name` as a prefix to the tar.gz file, and appends the extension. It appears to me that this new code would make the file have the extension duplicated.